genie-web/src/main/java/com/netflix/genie/web/services/JobDirectoryServerService.java

```diff
@@ -39,18 +41,24 @@
      * doesn't exist or an error is generated an appropriate HTTP error response will be written to {@code response}
      * instead.
      *
-     * @param jobId        The id of the job this request is for
-     * @param baseUrl      The base URL used to generate all URLs for resources
-     * @param relativePath The relative path from the root of the job directory of the expected resource
-     * @param request      The HTTP request containing all information about the request
-     * @param response     The HTTP response where all results should be written
+     * @param jobId            The id of the job this request is for
+     * @param baseUrl          The base URL used to generate all URLs for resources
+     * @param relativePath     The relative path from the root of the job directory of the expected resource
+     * @param request          The HTTP request containing all information about the request
+     * @param response         The HTTP response where all results should be written
+     * @param jobStatus        The current job status
+     * @param jobArchiveStatus The current archive status
+     * @param isV4             Whether an agent is executing this job
      * @throws GenieException If there is an error serving the response
      */
     void serveResource(
         String jobId,
         URL baseUrl,
         String relativePath,
         HttpServletRequest request,
-        HttpServletResponse response
+        HttpServletResponse response,
+        JobStatus jobStatus,
+        ArchiveStatus jobArchiveStatus,
+        boolean isV4
```

Comment:
I understand this optimization but I've avoided it as 1. it puts onus on calling services to know these pieces of information 2. Pushing the checks for them down as far as possible means we minimize the risk of looking in the wrong place due to a change in the time since original check and when we get here

Reply:
I see your point in (1), tho one could argue this is telling the service where to look / how to behave so it's not completely inappropriate.

But (2) I don't think is a valid reason. The calls in their original place happen 10 or so lines of java after where they are right now. That's probably microseconds.

Regardless. Got more changes coming to this so I may put them back where they were.